src/docfx/Models/FileMappingItem.cs

```diff
@@ -30,12 +30,24 @@ public class FileMappingItem
         [JsonProperty("files")]
         public FileItems Files { get; set; }
 
+        /// <summary>
+        /// The reference glob pattern collection, with path relative to property `src`/`cwd` is value is set
+        /// </summary>
+        [JsonProperty("references")]
+        public FileItems References { get; set; }
```

Comment:
Question: is this only needed when generating document from assembly? We don't need this when generating from source code?

Reply:
It can be possible to use it from sources as well. But the main reason is to use it for generating from assembly.

Reply:
Got your ideas. Looks good.

I think it may be better to move `references` from `metadata/src` to `metadata`. As you can see `metadata/src` is an array. If it has multiple objects, users may think `references` is needed for every object, as it looks like only applies to the current object in array. (Of course, I know it applies to the whole metadata section from your implementation). I'd expect the config like:
```
  "metadata": [
    {
      "src": [
        {
          "src": "../.tmp",
          "files": [
            "SomeLibrary.Common/lib/net462/*.dll"
          ],
        }
      ],
      "dest": "api/SomeLibrary.Common",
      "references": [
            "System.ValueTuple/lib/netstandard1.0/System.ValueTuple.dll"
          ]
    }
  ]
```
After this change `FileMappingItem`, we can stop glob support for it. User need to list all the required DLLs here. It will no be too long as far as we can see.

Reply:
@werwolfby Sounds nice. It makes sense to keep glob support after moving `references` out. 👍